File: BackEnd/PodcastProject/src/main/java/podcast/model/entities/dto/EpisodeDTO.java
package podcast.model.entities.dto;

import lombok.*;

import java.time.Duration;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class EpisodeDTO {
    private Integer id;
    private String title;
    private String description;
    private String audioPath;
    private String imageUrl;
    private Duration duration;
    private Integer views;
    private Integer season;
    private Integer chapter;
    private LocalDateTime publicationDate;
    private String podcastTitle;
}

File: FrontEnd/src/app/models/episode/episode-dto.ts
export interface EpisodeDTO {
  id: number;
  title: string;
  description: string;
  audioPath: string;
  imageUrl: string;
  duration: string; // Duration en formato ISO 8601 (ej: "PT1H30M")
  views: number;
  season: number;
  chapter: number;
  publicationDate: string; // LocalDateTime en formato ISO 8601
  podcastTitle: string;
}
